mllib/src/main/scala/org/apache/spark/ml/regression/LinearRegression.scala

```diff
@@ -186,7 +201,11 @@ class LinearRegression @Since("1.3.0") (@Since("1.3.0") override val uid: String
   override protected def train(dataset: Dataset[_]): LinearRegressionModel = {
     // Extract the number of features before deciding optimization solver.
     val numFeatures = dataset.select(col($(featuresCol))).first().getAs[Vector](0).size
-    val w = if (!isDefined(weightCol) || $(weightCol).isEmpty) lit(1.0) else col($(weightCol))
+    val w = if (!isDefined(weightCol) || $(weightCol).isEmpty) {
```

Comment:
we copy this code a lot. It would be nice to leave this as a common method, but since we've made it a shared param, there isn't really a way to add a method to the param to the `HasWeightCol` trait. Not sure if there is a nice way to do this. cc @holdenk